src/components/dialogs/contingency-list-selector.jsx

```diff
@@ -146,30 +141,25 @@ const ContingencyListSelector = (props) => {
     };
 
     const removeFromFavorite = (toRemove) => {
-        const toDelete = new Set(toRemove);
         saveFavorite(
             contingencyList
                 .map((e) => e.id)
-                .filter((item) => !toDelete.has(item))
+                .filter((item) => !toRemove.includes(item))
```

Comment:
why you dont filter directly like this contingencyList.filter((item) => !toRemoveSet.includes(item.id))

Reply:
saveFavorites expect only some ids, then we need to filter and map.